andela/travel_tool_front
# #160128175 Create user profile page

#### What does this PR do?
It creates the user profile page to allow users to add their profile information to the database.

#### Description of Task to be completed?
- create user profile tab on the settings page
- create tests for the user profile page
- add the user profile page
- connect it to the body shell
- Connect the frontend to the backend allowing users to add and get their user profile information

#### How should this be manually tested?
- Clone this repo - `git clone https://github.com/andela/travel_tool_front.git`
- Change into the repo directory - `cd travel_tool_front`
- Check out this branch - `git checkout ft-profile-page-settings-160128175`
- Install dependencies and start the server - `make start`
- Make sure that your backend server is running in the correct branch check (develop branch)
- Navigate to localhost:3000 in a browser
- Click the settings tab and access the user profile tab
- Fill the form with the correct credentials and click the save changes tab.
- Navigate to the request page and click the new request button to create a new request.
- The form should be prepopulated with your profile details.

#### Any background context you want to provide?
- Ensure the backend of the application is running on the develop branch
- Adding managers to the database so that users can add profile and travel requets. Go into the database and add a manager user, or just update your own account as a manager by updating your `roleId` in the database. You can find the `roleId`s in the `Roles` table or in the code inside the `seeders`.

#### What are the relevant pivotal tracker stories?
[#160128175](https://www.pivotaltracker.com/story/show/160128175)

#### Screenshots (if appropriate)
PROFILE PAGE TO UPDATE USERS PROFILE DETAILS TO THE DATABASE
<img width="1440" alt="screen shot 2018-09-17 at 22 14 27" src="https://user-images.githubusercontent.com/23402614/45645121-3332e880-bac8-11e8-92f1-dea50396dfb4.png">
AFTER SUCCESSFULLY UPDATING USER INFORMATION TO THE DATABASE
<img width="1440" alt="screen shot 2018-09-17 at 22 14 34" src="https://user-images.githubusercontent.com/23402614/45645125-3c23ba00-bac8-11e8-9b86-3516a5eeea4d.png">
FORM TO ADD TRAVEL REQUESTS UPDATED WITH THE NEW PROFILE DETAILS
<img width="696" alt="screen shot 2018-09-17 at 22 14 42" src="https://user-images.githubusercontent.com/23402614/45645135-45148b80-bac8-11e8-878c-38b896c1cf2a.png">
FORM TO ADD TRAVEL REQUESTS WITH CHECKBOX CLICKED MEANING THE USER WANTS TO ADD HIS DETAILS TO THE DATABASE
<img width="721" alt="screen shot 2018-09-18 at 09 49 17" src="https://user-images.githubusercontent.com/23402614/45669259-2ba32c00-bb28-11e8-8c25-21e052dface1.png">
ADD TRAVEL REQUEST WITH CHECKBOX ALSO UPDATES THE USER PROFILE DETAILS
<img width="1440" alt="screen shot 2018-09-18 at 09 49 28" src="https://user-images.githubusercontent.com/23402614/45669278-39f14800-bb28-11e8-8561-5458c09766e2.png">

#### Questions:
None
